### DISCUSSION TITLE
Deployment deletion does not delete pods

### DISCUSSION THREAD

**Discussion by dkoci** [okd-project/okd#1079] [Q&A]
Hi.
I dont know if this is OKD related issue, but anyway...
After upgrade to OKD 4.9 there is some really annoying feature enabled on all pods. All pods have ownerReference field which prevents pod deletion, before the deletion of its owner:
![image](https://user-images.githubusercontent.com/5565897/151168612-bfe51914-d113-45d5-94b0-4305103e108d.png)
The deletion of deployment does not delete pods, because they have owner reference to their replica-set. So it is required to manually delete replica-set.
Is there any way to disable this or any other workaround? All I want is that deletion of deployment delete all pods.
Tnx

**Comment by dkoci** [comment]
Anyone has the same problem?

**Comment by nate-duke** [comment]
@dkoci, it may help to see your Deployment/DeploymentConfig yaml. I've not seen anything like this in our okd4 clusters.

**Reply by dkoci** [reply]
I think it is kubernetes 1.22 related. The problem is in okd/openshift 4.9. Older releases works fine.
